chapter_01.md

```diff
@@ -1,2 +1,550 @@
+
+More importantly, Decision Made by the Board endpoint records a decision made by the board, but it also defines the following steps based on that decision. In real life, this decision will affect the course of action. Most of the time, these decision points and the actions derived by them are critical to run a cost-effective and efficient business. 
+https://github.com/salaboy/fmtok8s-c4p/blob/no-workflow/src/main/java/com/salaboy/conferences/c4p/C4PController.java#L60
+
+```
+@PostMapping(value = "/{id}/decision")
+    public void decide(@PathVariable("id") String id, @RequestBody ProposalDecision decision) {
+        emitEvent("> Proposal Approved Event ( " + ((decision.isApproved()) ? "Approved" : "Rejected") + ")");
+        Optional<Proposal> proposalOptional = proposalStorageService.getProposalById(id);
+        if (proposalOptional.isPresent()) {
+            Proposal proposal = proposalOptional.get();
+
+            // Apply Decision to Proposal
+            proposal.setApproved(decision.isApproved());
+            proposal.setStatus(ProposalStatus.DECIDED);
+            proposalStorageService.add(proposal);
+
+//          Only if it is Approved create a new Agenda Item into the Agenda Service
+            if (decision.isApproved()) {
+                agendaService.createAgendaItem(proposal);
+            }
+
+            // Notify Potential Speaker By Email
+            emailService.notifySpeakerByEmail(decision, proposal);
+        } else {
+            emitEvent(" Proposal Not Found Event (" + id + ")");
+        }
+
+    }
+
+```
+
+As you can probably guess, this method contains the logic of the entire flow. It starts by checking if the proposal Id can be found, if it is present it will apply the decision to the proposal object (Approved or Rejected) and then only if it was approved call the Agenda Service to create a new Agenda Item. No matter if it was accepted or not, an email needs to be sent to the potential speaker to notify him/her about the decision.  Both “agendaService” (https://github.com/salaboy/fmtok8s-c4p/blob/no-workflow/src/main/java/com/salaboy/conferences/c4p/services/AgendaService.java) and “emailService” (https://github.com/salaboy/fmtok8s-c4p/blob/no-workflow/src/main/java/com/salaboy/conferences/c4p/services/EmailService.java) are just encapsulating simple REST calls. 
+
+This simple method example, in real-life scenarios is never simple. Because of the inherent complexity of real-life challenges, you must expect simple methods like the one discussed above to become real monsters. The next section covers some of the major pitfalls and considerations that you must have in mind while writing business logic that is key to your domain. The next section also tries to share valuable resources, approaches, and projects which can help you to make batter decisions early on to avoid common mistakes.
+
+### Common Pitfalls
```

Comment:
I like the concepts presented here a lot.  I feel again, that we lost focus on the relation of all these great tips with how it relates to ddd concepts. Maybe we should think about moving part of the content here to a bonus section (I'm not sure if in this chapter, or a different one) and then link them here.
@otaviojava any thoughts?